docs/content/reference/ruby-go-changes.md

```diff
@@ -0,0 +1,105 @@
+---
+title: "Ruby to Go Migration"
+weight: 10
+---
+# What has changed in the MMv1 Go migration
+
+The Magic Modules code generator has been rewritten from Ruby to Go. For experienced contributors, this reference document lists what the expected changes are to the previous development workflow in Ruby.
+
+## YAML changes
+
+`.yaml` files within `mmv1/products` have had adjustments to the attribute typing. The initial Ruby lines `!ruby/object:Api::Type::<TYPE>` have been removed and replaced with a simpler `type: <TYPE>` line.
+
+Old Ruby YAML
+```yaml
+- !ruby/object:Api::Type::String
+  name: 'apiFieldName'
+  description: |
+    MULTILINE_FIELD_DESCRIPTION
+```
+
+New Go YAML
+```yaml
+- name: 'apiFieldName'
+  type: String
+  description: |
+    MULTI_LINE_FIELD_DESCRIPTION
+```
+
+## Template `.erb` file changes
+
+Template files have all been converted Embedded Ruby (ERB) templates to Go's [text/template](https://pkg.go.dev/text/template) format.
+All `.erb` files are replaced with equivalent `.tmpl` files. The MMv1 resource objects are passed to the Go templates for referencing, similar to the previous Ruby templates. For the list of available variables and functions within the templates, please reference:
+
+* [text/template standard library](https://pkg.go.dev/text/template)
+* [mmv1/api/resource.go](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/api/resource.go)
+* [mmv1/google/template_utils.go](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/google/template_utils.go)
+
+### Common templating snippets
+
+#### Version guards
+
+Old Ruby template
+```go
+<% unless version == 'ga' -%>
+// Go code here
+<% end -%>
+```
+
+New Go template
+```go
+{{ if ne $.TargetVersionName "ga" -}}
+// Go code here
+{{ end -}}
```

Comment:
```suggestion
{{- end }}
```
not a nitpick, since the the Go template closing whitespace eliminator tag is much more aggressive, people could easily delete several lines of whitespace or end up with other unexpected formatting if using this way for mid-file guards